src/swap/SwapReviewScreen.tsx

```diff
@@ -109,7 +102,7 @@ export function SwapReviewScreen(props: Props) {
           `Failure response fetching token swap quote. ${response.status}  ${response.statusText}`
         )
       }
-      setSwapInfo(await response.json())
+      setSwapInfo({ ...(await response.json()), userInput: params })
```

Comment:
why add params to the response? wouldn't this lead to inconsistencies (e.g. some parts of the response don't match the params and now we have `swapInfo` as a mixture of the 2)? i think we should find some way to avoid this, as this can lead to subtle and hard to debug bugs...

Reply:
Is pulling the userInput into the saga via selector preferred?

Reply:
Currently it looks like this:
```TypeScript
const json: SwapInfo = await response.json()
setSwapInfo(json)
...
// Dispatch swap submission
if (userInput !== null) {
  dispatch(swapStart({ ...swapInfo, userInput: userInput }))
}
```